framework/src/base/Assembly.C

```diff
@@ -1801,6 +1816,28 @@ Assembly::reinitFVFace(const FaceInfo & fi)
   _current_side_elem = _current_elem->build_side_ptr(_current_side).release();
 }
 
+QBase *
+Assembly::qruleFace(const Elem * face)
+{
+  // get all subdomain this face is a part of
+  std::set<SubdomainID> blocks;
+  for (subdomain_id_type i = 0; i < face->n_nodes(); i++)
+  {
+    auto & ids = _mesh.getNodeBlockIds(*(face->node_ptr(i)));
```

Comment:
If we pass in volumetric element and side, we do not need to call this costly function. I do not think we need to check all elements touching this side. We just need to check its two neighbors.

Reply:
Yeah.  This isn't a very good way - I agree.  I'll rework it to look at the neighboring elem instead of nodes.  This was just my lazy programmer approach trying to (re)use existing code.